features/steps/argonaut.py

```diff
@@ -344,3 +347,58 @@ def step_impl(context, field_name, value):
                 resource=res,
                 values=found,
                 value=value)
+
+
+def vital_unit_validation(field_name, resource, system_url):
+
+    path = field_name.split('.')
+    path.pop(0)
+
+    if path[0] == "valueQuantity":
```

Comment:
Consider collapsing these two branches by creating an intermediate list like `values_to_validate`, which is either  the list of components, or a a list with just the top-level value in it.